packages/graphql-server-micro/src/microApollo.ts

```diff
@@ -0,0 +1,75 @@
+import { GraphQLOptions, HttpQueryError, runHttpQuery } from 'graphql-server-core';
+import * as GraphiQL from 'graphql-server-module-graphiql';
+import { json } from 'micro';
+import * as url from 'url';
+import {IncomingMessage, ServerResponse} from 'http';
+
+export interface MicroGraphQLOptionsFunction {
+  (req?: IncomingMessage): GraphQLOptions | Promise<GraphQLOptions>;
+}
+
+export function microGraphql(options: GraphQLOptions | MicroGraphQLOptionsFunction) {
+  if (!options) {
+    throw new Error('Apollo Server requires options.');
+  }
+
+  if (arguments.length > 1) {
+    throw new Error(`Apollo Server expects exactly one argument, got ${arguments.length}`);
+  }
+
+  return async function (req: IncomingMessage, res: ServerResponse) {
+    let query;
+    if (req.method === 'POST') {
+      try {
+        query = await json(req);
+      } catch (err) {
+        query = undefined;
+      }
+    } else {
+      query = url.parse(req.url, true).query;
+    }
+
+    runHttpQuery([req, res], {
+      method: req.method,
+      options: options,
+      query: query,
+    }).then((gqlResponse) => {
+      res.setHeader('Content-Type', 'application/json');
+      res.write(gqlResponse);
+      res.end();
+    }, (error: HttpQueryError) => {
+      if ( 'HttpQueryError' !== error.name ) {
+        throw error;
+      }
+
+      if ( error.headers ) {
```

Comment:
Is this something we added recently? I don't remember having this in the initial versions. What are people using those headers for?